### PR TITLE
fix: make @cap-js/sqlite work with better-sqlite3@9.3.0

### PR DESCRIPTION
**Pull request by BobdenOs**
With the release of `better-sqlite3@9.3.0` it bumped the sqlite version to `3.45.0`. Which comes with a complete rework of the `json` functions. With the big difference being that it now supports an internal `jsonb` format. Which is supposed to be only used by `sqlite` itself, but with this new change all the `json` functions accept `binary` data as the `jsonb` format. Therefor as the json that we produced was being converted into `Buffer`s it resulted in `better-sqlite3` providing them as `binaries` to the `json_each` function which made it interoperate it as an malformed `jsonb`. So this change forced the `json` to be bound as a `strings` which makes everything in its own behavior.

Additionally now that the `json_insert` function has a `jsonb_insert` equavelant it fixes a memory issue `sqlite` had introduced in the past and we can remove the `json_merge` custom function again. Which removes the requirement to call back to `js` when selecting more then `50` columns. Which should make reading many columns faster again.

### CODE REVIEW

File: db-service/lib/cqn2sql.js [hunk 1]
```diff
@@ -252,21 +252,14 @@ class CQN2SQLRenderer {
...
-      obj = `json_merge(${chunks})`
+    // Prevent SQLite from hitting function argument limit of 100
+    let obj = "'{}'"
+    for (let i = 0; i < cols.length; i += 48) {
+      obj = `json${isRoot ? '' : 'b'}_insert(${obj},${cols.slice(i, i + 48)})`
     }
-
-
-    return `SELECT ${SELECT.one || SELECT.expand === 'root' ? obj : `json_group_array(${obj.includes('json_merge') ? `json_insert(${obj})` : obj})`} as _json_ FROM (${sql})`
+    return `SELECT ${isRoot || SELECT.one ? obj : `jsonb_group_array(${obj})`} as _json_ FROM (${sql})`
```
Comment (patricebender) [inline db-service/lib/cqn2sql.js:262]: just wondering about the sqlite specific stuff in the generic `db-service`, could you elaborate why we need this here?

**Review by danjoa — APPROVED** [review]
Looks good to me

File: db-service/lib/cqn2sql.js (hunk 1 above)
Comment (BobdenOs) [reply db-service/lib/cqn2sql.js]: It has always been in the generic `db-service`. The original reason for having it here is that there is no `SQL` specified uniform `API` to create `json` values. Therefor taking `SQLite` makes it easy for other database services to find their equivalent functions.

Comment (patricebender) [reply db-service/lib/cqn2sql.js]: Got it. Thanks!

**Pull request merged**